godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/Plane.kt

```diff
@@ -0,0 +1,261 @@
+     * If no intersection is found, null is returned.
+     */
+    fun intersect3(plane1: Plane, plane2: Plane): Vector3? {
+        val plane0 = this
+        val normal0 = plane0.normal
+        val normal1 = plane1.normal
+        val normal2 = plane2.normal
+
+        val denom = normal0.cross(normal1).dot(normal2)
+
+        if (abs(denom) <= CMP_EPSILON) {
+            return null
+        }
+
+        val result: Vector3?
+        result = (normal1.cross(normal2) * plane0.d) +
+            (normal2.cross(normal0) * plane1.d) +
+            (normal0.cross(normal1) * plane2.d) / denom
+        return result
+    }
+
+    /**
+     * Returns the intersection point of a ray consisting of the position from and the direction normal dir with this plane.
+     * If no intersection is found, null is returned.
+     */
+    fun intersectsRay(from: Vector3, dir: Vector3): Vector3? {
+        val den = normal.dot(dir)
+
+        if (abs(den) <= CMP_EPSILON) {
+            return null
+        }
+
+        val dist = (normal.dot(from) - d) / den
+
+        if (dist > CMP_EPSILON) {
+            return null
+        }
+
+        return from + dir * (-dist)
+    }
+
+    /**
+     * Returns the intersection point of a segment from position begin to position end with this plane.
+     * If no intersection is found, null is returned.
+     */
+    fun intersectsSegment(p_begin: Vector3, p_end: Vector3): Vector3? {
+        val segment = p_begin - p_end
+        val den = normal.dot(segment)
+
+        if (abs(den) <= CMP_EPSILON) {
+            return null
+        }
+
+        val dist = (normal.dot(p_begin) - d) / den
+
+        if (dist < -CMP_EPSILON || dist > (1.0 + CMP_EPSILON)) {
+            return null
+        }
+
+        return p_begin + segment * (-dist)
+    }
+
+    /**
+     * Returns true if this plane and plane are approximately equal, by running isEqualApprox on each component.
+     */
+    fun isEqualApprox(other: Plane): Boolean {
+        return this.normal.isEqualApprox(other.normal) && isEqualApprox(this.d, other.d)
+    }
+
+    /**
+     * Returns true if point is located above the plane.
+     */
+    fun isPointOver(point: Vector3): Boolean {
+        return normal.dot(point) > d
+    }
+
+    /**
+     * Returns a copy of the plane, normalized.
+     */
+    fun normalized(): Plane {
+        val p = this
+        p.normalize()
+        return p
+    }
+
+    internal fun normalize() {
+        val l = normal.length()
+        if (l == 0.0) {
+            this.normal = Vector3()
+            this.d = 0.0
+            return
+        }
+        normal /= l
+        d /= l
+    }
+
+    /**
+     * Returns the orthogonal projection of point p into a point in the plane.
+     */
+    fun project(point: Vector3): Vector3 {
+        return point - normal * distanceTo(point)
+    }
+
+
+    //UTILITIES
+    override fun toString(): String {
+        return "${this.normal.toString()},$d"
```

Comment:
same as the comment above, `Plane(normal=<normal>, d=<d>)`

Reply:
same as previous comment

Reply:
Also agree with @raniejade